cmd/podman/machine/machine.go

```diff
@@ -101,7 +99,21 @@ func autocompleteMachine(_ *cobra.Command, args []string, toComplete string) ([]
 	if len(args) == 0 {
 		return getMachines(toComplete)
 	}
-	return nil, cobra.ShellCompDirectiveNoFileComp
+	return nil, cobra.ShellCompDirectiveNoSpace
+}
+
+func autocompleteMachineProvider(_ *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	var suggestions []string
+	for _, p := range provider2.GetAll() {
+		if len(toComplete) > 0 {
+			if strings.HasPrefix(p.VMType().String(), toComplete) {
+				suggestions = append(suggestions, p.VMType().String())
+			}
+		} else {
+			suggestions = append(suggestions, p.VMType().String())
+		}
```

Comment:
you don't have to filter on toComplete at all this can be just be
```
suggestions = append(suggestions, p.VMType().String())
```

which would be simpler here, the shells already know to ignore non matching results